core/src/main/java/org/mobilitydata/gtfsvalidator/parsing/RowParser.java

```diff
@@ -216,6 +217,18 @@ public String asText(int columnIndex, boolean required) {
 
   @Nullable
   public String asId(int columnIndex, boolean required) {
+    String value = row.asString(columnIndex);
+    if (value == null) {
+      return asString(columnIndex, required);
+    }
+    for (char ch : value.toCharArray()) {
+      if (!(ch >= 32 && ch < 127)) {
+        addErrorInRow(
+            new NonAsciiOrNonPrintableCharNotice(
+                row.getFileName(), row.getRowNumber(), row.getColumnName(columnIndex)));
+        break;
+      }
+    }
```

Comment:
Why are you calling asString once again at the end if you already have the value to return?